src/controllers/UserController.ts

```diff
@@ -0,0 +1,281 @@
+import { Request as req } from 'express';
+import logger from '../middleware/logger';
+import { getUserList, deactivateUsers } from '../db/Users.db';
+import { TypeORMError } from 'typeorm';
+import { authenticate } from '../middleware/authentication';
+import { AuthenticationError } from '../middleware/AuthenticationError';
+
+@Middlewares(authenticate)
+@Route('users')
+export class UserController extends Controller {
+    /*
+     * Use to get user list based on status
+     * Expected error codes and messages:
+     * - 200
+     * -- 'OK'
+     * - 204
+     *  -- 'No Content'
+     * - 400
+     *  -- 'Bad Request'
+     * - 401
+     * -- 'Unauthorized'
+     * - 403
+     * -- 'Forbidden'
+     * - 404
+     * -- 'Not Found'
+     * @param active Status of the user
+     * @returns A list of users based on active status
+     */
+
+    @Get('')
+    public async getAllUsers(
+        @Res() unauthorizedErrorResponse: TsoaResponse<401, unauthorizedError>,
+        @Res() badRequestErrorResponse: TsoaResponse<400, badRequestError>,
+        @Res() forbiddenErrorResponse: TsoaResponse<403, forbiddenError>,
+        @Res() notFoundErrorResponse: TsoaResponse<404, notFoundError>,
+        @Res() serverErrorResponse: TsoaResponse<500, serverErrorType>,
+        @Res()
+        noActiveFoundResponse: TsoaResponse<204, noActiveUserFound>,
+        @Query() active: string,
+        @Request() req: req,
+    ): Promise<Array<userList>> {
+        let results: Array<userList> = [];
+        let permissions: string[] = [];
+        // checking permissions for this api.
+        try {
+            permissions = decodingJWT(req.cookies.token)?.payload.permissions;
+            if (!permissions.includes('USER_ACCESS')) {
+                throw new AuthenticationError(
+                    `Permission 'USER_ACCESS' is not available for this user`,
+                    403,
+                );
+            }
+        } catch (err) {
+            if (err instanceof AuthenticationError) {
+                logger.warn(
+                    `Encountered 403 forbidden error in getAllUsers: ${err.message}`,
+                );
+                return forbiddenErrorResponse(403, {
+                    message: err.message,
+                    code: 403,
+                });
+            }
+            if (err instanceof Error) {
+                logger.warn(
+                    `Encountered 404 not found error in getAllUsers: ${err.message}`,
+                );
+                return notFoundErrorResponse(404, {
+                    message: err.message,
+                    code: 404,
+                });
+            }
+        }
+        // retrieving data for a particular status
+        try {
+            let where;
+            let userStatus;
+            if (active === 'true') {
+                where = { isActive: true };
+                userStatus = 'active';
+            } else if (active === 'false') {
```

Comment:
This is query a parameter, so what happens if it is specified incorrectly or not specified? Wouldn't it just show all users?